state/mysql/mysql.go

```diff
@@ -305,6 +371,44 @@ func (m *MySQL) ensureStateTable(ctx context.Context, stateTableName string) err
 		}
 	}
 
+	// Check if expiredate column exists - to cater cases when table was created before v1.11.
+	columnExists, err := columnExists(ctx, m.db, schemaName, stateTableName, "expiredate", m.timeout)
```

Comment:
there's a potential for a race condition here if 2 sidecars are starting up at the same time.

Because you check for the column's existence and then after that you add the column, it's possible that 2 sicecars starting up at the same time will both detect the column doesn't exist, and will both try to create that.

In the Postgres component, we solve that by acquiring a lock (not using transactions because, at least in Postgres, table creation doesn't happen in a transaction). I don't know if MySQL (and MariaDB) support something like that.

Worst case, I guess, one of the two sidecars will fail to initialize and will need to be restarted? This should only happen during upgrades from <1.11 anyways

Reply:
I have added `IF NOT EXISTS` to column and index addition - that way, in case of such a race condition, only first one will be able to add column and index and other one will NOT fail, it will simply skip past. Hope it suffices.

Reply:
That should do it, thanks